pulp_rpm/app/modulemd.py

```diff
@@ -50,130 +58,124 @@ def modules_packages(modules):
     version.add_content(Package.objects.filter(pk__in=packages_to_add))
 
 
-def parse_modulemd(module_names, module_index):
+def split_modulmd_file(file):
     """
-    Get modulemd NSVCA, artifacts, dependencies.
-
-    Args:
-        module_names (list):
-            list of modulemd names
-        module_index (mmdlib.ModuleIndex):
-            libmodulemd index object
-
-    Returns:
-        list of modulemd as dict
-
+    Helper method to preserve original formatting of modulemd.
     """
-    ret = list()
-    for module in module_names:
-        for stream in module_index.get_module(module).get_all_streams():
-            modulemd = dict()
-            modulemd[PULP_MODULE_ATTR.NAME] = stream.props.module_name
-            modulemd[PULP_MODULE_ATTR.STREAM] = stream.props.stream_name
-            modulemd[PULP_MODULE_ATTR.VERSION] = str(stream.props.version)
-            modulemd[PULP_MODULE_ATTR.STATIC_CONTEXT] = stream.props.static_context
-            modulemd[PULP_MODULE_ATTR.CONTEXT] = stream.props.context
-            modulemd[PULP_MODULE_ATTR.ARCH] = stream.props.arch
-            modulemd[PULP_MODULE_ATTR.ARTIFACTS] = stream.get_rpm_artifacts()
-            modulemd[PULP_MODULE_ATTR.DESCRIPTION] = stream.get_description()
-
-            modulemd[PULP_MODULE_ATTR.PROFILES] = {}
-            for profile in stream.get_profile_names():
-                stream_profile = stream.get_profile(profile)
-                modulemd[PULP_MODULE_ATTR.PROFILES][profile] = stream_profile.get_rpms()
-
-            dependencies_list = stream.get_dependencies()
-            dependencies = list()
-            for dep in dependencies_list:
-                depmodule_list = dep.get_runtime_modules()
-                platform_deps = dict()
-                for depmod in depmodule_list:
-                    platform_deps[depmod] = dep.get_runtime_streams(depmod)
-                dependencies.append(platform_deps)
-            modulemd[PULP_MODULE_ATTR.DEPENDENCIES] = dependencies
-            # create yaml snippet for this modulemd stream
-            temp_index = mmdlib.ModuleIndex.new()
-            temp_index.add_module_stream(stream)
-            modulemd["snippet"] = temp_index.dump_to_string()
-            ret.append(modulemd)
-    return ret
-
-
-def parse_defaults(module_index):
+    m_open = (
+        gzip.open if file.url.endswith(".gz") else lzma.open if file.url.endswith(".xz") else open
```

Comment:
We should double check that there are no plans to use zchunk compression: https://github.com/fedora-modularity/libmodulemd/blob/83f7924f61c0e24263aa92ac392b2430501d3b32/modulemd/include/modulemd-2.0/modulemd-compression.h#L57

And if they were to do that for some stupid reason (please no), what would be our options?

Reply:
unfortunately options are to re-use `libmodulemd` or to decompress it by system call as `zck` doesn't have python extension yet

Reply:
Recap: should be fine, the data should never be `_zck` compressed without a `_zck` suffix in the repomd.xml, and we ignore those.

Reply:
added check before `modulemd` parsing as they go first, if `zck` occurres sync will fail